components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -2652,6 +2653,95 @@ bool JsonRpcService::AddSwitchEthereumChainRequest(const std::string& chain_id,
   return true;
 }
 
+void JsonRpcService::GetEthTokenSymbol(const std::string& contract_address,
+                                       const std::string& chain_id,
+                                       GetEthTokenSymbolCallback callback) {
+  auto network_url = GetNetworkURL(prefs_, chain_id, mojom::CoinType::ETH);
+  if (!network_url.is_valid()) {
+    std::move(callback).Run(
+        "", mojom::ProviderError::kInvalidParams,
+        l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+    return;
+  }
+  const std::string data = GetFunctionHash("symbol()");
+  auto internal_callback =
+      base::BindOnce(&JsonRpcService::OnGetEthTokenSymbol,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+  RequestInternal(eth::eth_call("", contract_address, "", "", "", data,
+                                kEthereumBlockTagLatest),
+                  true, network_url, std::move(internal_callback));
+}
+
+void JsonRpcService::OnGetEthTokenSymbol(GetEthTokenSymbolCallback callback,
+                                         APIRequestResult api_request_result) {
+  if (!api_request_result.Is2XXResponseCode()) {
+    std::move(callback).Run(
+        "", mojom::ProviderError::kInternalError,
+        l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR));
+    return;
+  }
+
+  std::string symbol;
+  if (!eth::ParseStringResult(api_request_result.value_body(), &symbol)) {
+    mojom::ProviderError error;
+    std::string error_message;
+    ParseErrorResult<mojom::ProviderError>(api_request_result.value_body(),
+                                           &error, &error_message);
+    std::move(callback).Run("", error, error_message);
+    return;
+  }
+  std::move(callback).Run(symbol, mojom::ProviderError::kSuccess, "");
+}
+
+void JsonRpcService::GetEthTokenDecimals(const std::string& contract_address,
+                                         const std::string& chain_id,
+                                         GetEthTokenDecimalsCallback callback) {
+  auto network_url = GetNetworkURL(prefs_, chain_id, mojom::CoinType::ETH);
+  if (!network_url.is_valid()) {
+    std::move(callback).Run(
+        "", mojom::ProviderError::kInvalidParams,
+        l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+    return;
+  }
+  const std::string data = GetFunctionHash("decimals()");
+  auto internal_callback =
+      base::BindOnce(&JsonRpcService::OnGetEthTokenDecimals,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+  RequestInternal(eth::eth_call("", contract_address, "", "", "", data,
+                                kEthereumBlockTagLatest),
+                  true, network_url, std::move(internal_callback));
+}
```

Comment:
Note - Jocelyn had asked that GetEthTokenSymbol and GetEthTokenDecimals share the code since they are similar, however because `symbol()`and `decimals()` return different types (string and uint8  respectively), their parsing logic is different, so I didn't see much opportunity for reuse.